pandas/core/common.py

```diff
@@ -479,6 +481,43 @@ def _dict_compat(d):
                 for key, value in iteritems(d))
 
 
+def _standardize_mapping(into):
+    """
+    Helper function to standardize the supplied mapping so it can
+    be passed to the ``Series.to_dict()`` and ``DataFrame.to_dict()``
+
+    Parameters
+    ----------
+    into : instance or subclass of collections.Mapping
+        The argument supplied to ``to_dict``.  Must be a class, an
+        initialized collections.defaultdict, or an empty instance
+        of a collections.Mapping subclass.
+
+    Returns
+    -------
+    mapping : a collections.Mapping subclass or other constructor
+        a callable object that can accept an iterator to create
+        the desired Mapping.
+
+    """
+    if not inspect.isclass(into):
+        if len(into) > 0:
+            raise ValueError(
+                "to_dict() only accepts empty mappings.")
+        elif type(into) == collections.defaultdict:
+            return partial(
+                collections.defaultdict, into.default_factory)
+        else:
```

Comment:
For the ``return partial(...)`` part, I'm trying to get around the way `defaultdict` accepts arguments.  The other out-of-the-box mappings will happily accept only an iterator, which is what both of the `to_dict()` functions supply to the return value of this function.  ``defaultdict`` also needs a `default_factory`, so I use `partial` to include the `default_factory` so the `to_dict()` functions can supply only an iterator.  

For the last ``else``, I'm just sending the class of the supplied mapping (once its been confirmed empty) back to the function to avoid duplicated logic.  

If you think a different structure works better, or it should be broken up into multiple functions, let me know.